c7n/resources/s3.py

```diff
@@ -789,6 +1029,212 @@ def process_bucket(self, func, bucket, topic, account_id, session_factory):
         return source.add(func)
 
 
+@actions.register('configure-lifecycle')
+class ConfigureLifecycle(BucketActionBase):
+    """Action that aggregates all rules with whole bucket prefix and
+    changes them to the configured policy values."""
+
+    permissions = ('s3:GetLifecycleConfiguration', 's3:PutLifecycleConfiguration', )
+
+    schema = {
+        'type': 'object',
+        'additionalProperties': True,
+        'properties': {
+            'type': {'enum': ['configure-lifecycle']},
+            'id': {'type': 'string'},
+            'prefix': {'type': 'string'},
+            'deletes_objects': {'type': 'boolean'},
+            'delete_days': {'type': 'number'},
+            'multipart_days': {'type': 'number'},
+            'ia_days': {'type': 'number'},
+            'glacier_days': {'type': 'number'},
+            'max_workers': {'type': 'number'}
+        }
+    }
+
+    cache_update = False
+
+    def process(self, buckets):
+            max_workers = self.data.get('max_workers', DEFAULT_WORKERS)
+
+            with self.executor_factory(max_workers=max_workers) as w:
+                saved_buckets = w.map(self.call_api, buckets)
+                saved_buckets = filter(None, list(saved_buckets))
+
+            if self.cache_update:
+                saved_buckets = {b['Name']: b for b in saved_buckets}
+                self.manager._cache.save(CACHE_NAME, saved_buckets)
```

Comment:
I think the caching is good - I was just wondering if we can cache after applying the lifecycle.  When I was testing my policy I was repeatedly applying a lifecycle to the same resource because it kept loading the resource from the cache and the cached version didn't have the lifecycle applied.  I think this would also mean that the `has-lifecycle` filter would not catch that resource until the cached copy expired.